apps/api/src/app/step-schemas/dtos/step-schema.dto.ts

```diff
@@ -1,6 +1,83 @@
 import { JSONSchema } from 'json-schema-to-ts';
+import { StepType } from '@novu/shared';
+
+import { ActionStepEnum, ChannelStepEnum } from '@novu/framework/internal';
+
+const chatProperties = {
+  body: { type: 'chatBody' },
+};
+
+const emailProperties = {
+  subject: { type: 'emailSubject' },
+  body: { type: 'emailBody' },
+};
+
+const inAppProperties = {
+  subject: { type: 'inAppSubject' },
+  body: { type: 'inAppBody' },
+  avatar: { type: 'inAppAvatar' },
+  primaryAction: { type: 'inAppPrimaryAction' },
+  secondaryAction: { type: 'inAppSecondaryAction' },
+  data: { type: 'inAppData' },
+  redirect: { type: 'inAppRedirect' },
+};
+
+const pushProperties = {
+  subject: { type: 'pushSubject' },
+  body: { type: 'pushBody' },
+};
+
+const smsProperties = {
+  body: { type: 'smsBody' },
+};
+
+const delayProperties = {
+  type: { type: 'delayType' },
+  amount: { type: 'delayAmount' },
+  unit: { type: 'delayUnit' },
+};
+
+const digestProperties = {
+  amount: { type: 'digestAmount' },
+  unit: { type: 'digestUnit' },
+  digestKey: { type: 'digestKey' },
+  lookBackWindow: { type: 'digestLookBackWindow' },
+};
+
+const customProperties = {};
+
+type Properties =
+  | typeof chatProperties
+  | typeof emailProperties
+  | typeof inAppProperties
+  | typeof pushProperties
+  | typeof smsProperties
+  | typeof delayProperties
+  | typeof digestProperties
+  | typeof customProperties;
+
+export type UiSchema = {
```

Comment:
We can use discriminated unions and drastically simplify the type definition. See [here](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes-func.html#discriminated-unions).

That is we create a union for each step type and we don't need the map anymore.